applications/ChimeraApplication/custom_strategies/custom_builder_and_solver/residualbased_block_builder_and_solver_with_constraints_for_chimera.h

```diff
@@ -0,0 +1,312 @@
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class ResidualBasedBlockBuilderAndSolverWithConstraintsForChimera
+ * @ingroup KratosChimeraApplication
+ * @brief Current class provides an implementation for applying the chimera constraints that is enforcing continuity.
+ * @details This implementation enforces continuity necessary for chimera in the following way :
+ *
+ *
+ *
+ *   L =        [I  0  0 ]
+ *              [0  I  0 ]
+ *
+ *  K_mod = L'KT
+ *  F_mod = L'(F-K*g)
+ * 
+ *  Where T has the same definition as that of the classical master slave constraints
+ *
+ * @author Aditya Ghantasala
+ */
+template <class TSparseSpace,
+          class TDenseSpace,
+          class TLinearSolver>
+class KRATOS_API(CHIMERA_APPLICATION) ResidualBasedBlockBuilderAndSolverWithConstraintsForChimera
+    : public ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Definition of the base class
+    typedef ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver> BaseType;
+
+    // The size_t types
+    typedef typename BaseType::IndexType IndexType;
+
+    /// Definition of the classes from the base class
+    typedef typename BaseType::TSchemeType TSchemeType;
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+
+
+    KRATOS_CLASS_POINTER_DEFINITION(ResidualBasedBlockBuilderAndSolverWithConstraintsForChimera);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /** Constructor.
+     */
+    explicit ResidualBasedBlockBuilderAndSolverWithConstraintsForChimera(
+        typename TLinearSolver::Pointer pNewLinearSystemSolver)
+        : ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver>(pNewLinearSystemSolver)
+    {
+    }
+
+    /** Destructor.
+     */
+    ~ResidualBasedBlockBuilderAndSolverWithConstraintsForChimera() override
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+
+    /**
+     * @brief This function is intended to be called at the end of the solution step to clean up memory storage not needed
+     */
+    void Clear() override
+    {
+        BaseType::Clear();
+        mL.resize(0,0,false);
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "ResidualBasedBlockBuilderAndSolverWithConstraintsForChimera";
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+    TSystemMatrixType mL; /// This is L matrix described above (at class definition)
```

Comment:
As far as I understand, this is the matrix to enforce the continuity in the solution between the patches and the background mesh. Hence I assume that it is "small" compared to the global system matrix one, so it might be not an issue to store it. Am I wrong?

Besides, are the constraints always fixed? Shouldn't this matrix be recomputed as the patches move across the domain?

Reply:
mL is same size as that of mT. 

And yes it will change as the patch moves across the background.